CONTRIBUTING.md

```diff
@@ -0,0 +1,177 @@
+ We do not enforce a naming convention for the PRs, but **please use something descriptive of your changes**, keeping in mind that the title of your PR will be automatically added to the following [release changelog](https://github.com/meilisearch/cloud-providers/releases/).
+
+You can check out the more extended, complete guideline documentation [here](https://github.com/meilisearch/.github/blob/main/Hacktoberfest_2022_contributors_guidelines.md).
+
+## Development Workflow
+
+### Setup <!-- omit in toc -->
+
+1. Install the [Packer CLI](https://developer.hashicorp.com/packer/downloads)
+2. Set up all your needed credentials
+- Aws:
+Obtain your [AWS access keys](https://docs.aws.amazon.com/keyspaces/latest/devguide/access.credentials.html) and set it in your environment:
+  ``` bash
+    export AWS_ACCESS_KEY_ID="YOUR_ACCESS_KEY"
+    export AWS_SECRET_ACCESS_KEY="YOUR_SECRET_KEY"
+  ```
+- DigitalOcean:
+Before running any script, make sure to [obtain a DigitalOcean API Token](https://www.digitalocean.com/docs/apis-clis/api/create-personal-access-token/) and set it in your environment:
+  ```bash
+  export DIGITALOCEAN_ACCESS_TOKEN="XxXxxxxXXxxXXxxXXxxxXXXxXxXxXX"
+  ```
+- Gcp
+Obtain your [GCP credentials](https://cloud.google.com/docs/authentication/getting-started) and set it in your environment:
+  ```bash
+  export GOOGLE_APPLICATION_CREDENTIALS="path_to_your_creadential_file.json"
+  ```
+3. Download and install Packer plugins
+    ``` bash
+    packer init .
+    ```
+
+### Tests and Linter <!-- omit in toc -->
+
+Each PR should pass the tests and the linter to be accepted.
+
+Use this command to format HCL2 configuration file:
+``` bash
+packer fmt .
+```
+
+Use this command to validate the syntax and configuration of the template:
+``` bash
+packer validate .
+```
+
+### Build the images <!-- omit in toc -->
+
+⚠ Please be aware that this command will create all new Meilisearch images on all platforms.
+
+This command will execute the template and build all images in parallel:
+``` bash
+packer build meilisearch.pkr.hcl
+```
+
+You can build one or more images by specifying their name, take the first parameter given to `source` block and add it the following wildcard regex: `.*`.
+Example:
+``` bash
+packer build -only 'amazon-ebs.*' .
+```
+
+### Add a new provider
+
+There are multiple providers available for Packer, you can check the [External Plugins Documentation](https://developer.hashicorp.com/packer/plugins)
+
+To add a provider correctly to the list of existing providers, proceed as follows:
+1. Add your provider's code to the main `meilisearch.pkr.hcl` file as expected.
+2. Add the provider name to the [README](README.md#-providers-available).
+3. Add a manual trigger for your new provider to the CI by adding it to the list of suppliers in the `option` block of [manual_publish.yml](/manual_publish.yml).
+
```

Comment:
I think the first one should be:

1. Explain through detailed instructions using text/images/video the steps you've used to create and configure the provider to be able to receive new images.